packages/manager/modules/common-api/src/services/hooks/useDeleteService.ts

```diff
@@ -0,0 +1,47 @@
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+  getResourceServiceId,
+  getResourceServiceIdQueryKey,
+  deleteService,
+} from '../api';
+
+export type DeleteServiceMutationParams = {
+  resourceName: string;
+};
+
+export const deleteServiceMutationKey = ['delete-service'];
+
+export type UseDeleteServiceParams = {
+  onSuccess?: () => void;
+  onError?: (result: ApiError) => void;
+  mutationKey?: string[];
+};
+
+export const useDeleteService = ({
+  onSuccess,
+  onError,
+  mutationKey = deleteServiceMutationKey,
+}: UseDeleteServiceParams) => {
+  const queryClient = useQueryClient();
+  const { mutate: terminateService, ...mutation } = useMutation({
+    mutationKey,
+    mutationFn: async ({ resourceName }: DeleteServiceMutationParams) => {
+      const { data } = await queryClient.fetchQuery<
+        ApiResponse<number[]>,
+        ApiError
+      >({
+        queryKey: getResourceServiceIdQueryKey({ resourceName }),
+        queryFn: () => getResourceServiceId({ resourceName }),
+      });
+      return deleteService({ serviceId: data[0] });
+    },
+    onSuccess,
+    onError,
+  });
```

Comment:
Same, I this just options type for extends the useMutations params is more maintainable

Reply:
```suggestion
export const useDeleteService = ({
  mutationKey,
  ...options
}: Partial<
  MutationOptions<
    AxiosResponse<
      {
        message: string;
      },
      unknown
    >,
    ApiError,
    DeleteServiceMutationParams
  >
>) => {
  const queryClient = useQueryClient();
  const { mutate: terminateService, ...mutation } = useMutation({
    mutationKey: mutationKey ?? deleteServiceMutationKey,
    mutationFn: async ({ resourceName }: DeleteServiceMutationParams) => {
      const { data } = await queryClient.fetchQuery<
        ApiResponse<number[]>,
        ApiError
      >({
        queryKey: getResourceServiceIdQueryKey({ resourceName }),
        queryFn: () => getResourceServiceId({ resourceName }),
      });
      return deleteService({ serviceId: data[0] });
    },
    ...options,
  });
```

Reply:
done thanks